exchange/bidder.go

```diff
@@ -327,23 +327,39 @@ func getAssetByID(id int64, assets []nativeRequests.Asset) (nativeRequests.Asset
 
 // makeExt transforms information about the HTTP call into the contract class for the PBS response.
 func makeExt(httpInfo *httpCallInfo) *openrtb_ext.ExtHttpCall {
+	if httpInfo == nil {
+		return &openrtb_ext.ExtHttpCall{}
+	}
+
 	if httpInfo.err == nil {
-		return &openrtb_ext.ExtHttpCall{
-			Uri:            httpInfo.request.Uri,
-			RequestBody:    string(httpInfo.request.Body),
-			ResponseBody:   string(httpInfo.response.Body),
-			Status:         httpInfo.response.StatusCode,
-			RequestHeaders: httpInfo.request.Headers,
+		if httpInfo.request != nil && httpInfo.response != nil {
+			return &openrtb_ext.ExtHttpCall{
+				Uri:            httpInfo.request.Uri,
+				RequestBody:    string(httpInfo.request.Body),
+				ResponseBody:   string(httpInfo.response.Body),
+				Status:         httpInfo.response.StatusCode,
+				RequestHeaders: filterHeader(httpInfo.request.Headers),
+			}
 		}
-	} else if httpInfo.request == nil {
-		return &openrtb_ext.ExtHttpCall{}
 	} else {
-		return &openrtb_ext.ExtHttpCall{
-			Uri:            httpInfo.request.Uri,
-			RequestBody:    string(httpInfo.request.Body),
-			RequestHeaders: httpInfo.request.Headers,
+		if httpInfo.request != nil {
+			return &openrtb_ext.ExtHttpCall{
+				Uri:            httpInfo.request.Uri,
+				RequestBody:    string(httpInfo.request.Body),
+				RequestHeaders: filterHeader(httpInfo.request.Headers),
+			}
 		}
 	}
+
+	return &openrtb_ext.ExtHttpCall{}
+}
+
+var authorizationHeader = http.CanonicalHeaderKey("authorization")
+
+func filterHeader(h http.Header) map[string][]string {
+	clone := h.Clone()
```

Comment:
Wondering if we need to clone the http header map here? Do we expect the authorization header to be used anywhere else outside of this function? If so then it makes sense to not modify the original object but if not then we can just delete it from the original object

Reply:
I checked the code and I don't see any other usage of the Authorization header so it should be safe to remove it from the original object here